### PR TITLE
Recipe App

### PR DESCRIPTION
**Pull request by MrBrN197**
### Changes | 🏁

- Login page and registration page:
    - Should be built with Devise.
- Food list:
    - Should display a list of food as in the wireframe.
    - Should lead to the form that allows users to add new food.
- Recipies list:
    - Should display a list of recipes created by the logged-in user as in the wireframe.
    - Should lead to recipe details.
    - If the user is the owner of the recipe, should allow user to delete it.
- Public recipe list:
    - Should display a list of all public recipes ordered by newest as in the wireframe.
    - Should lead to recipe details.
- Recipe details:
    - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
    - If the user is the owner of the recipe, should lead to the form that allows user to add new food.
- General shopping list view (**do not implement if in group of 3**):
    - shows the list of food that is missing for all recipes of the logged-in user.
    - counts the total food items and total price of the missing food.
- Handle N+1 queries.
- Created navigation menu that allows users to open all of the pages you created.

### CODE REVIEW

**Comment by MrBrN197** [comment]
Please note that the `Ingredient` model will map the `foods_recipes` table from the ERD diagram. any code that references the word `ingredient` is referring to the `FoodRecipe` ERD model. we renamed this simply for convenience.

File: app/controllers/foods_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  def create
+    new_food = current_user.foods.new(food_params)
+    if new_food.save
+      flash[:success] = 'You successfully added a food item'
+      redirect_to foods_url
+    else
+      flash.now[:error] = 'Error: You could not add a food item'
+      render :new
+    end
```
Comment (rloterh) [inline app/controllers/foods_controller.rb:18]: - [x] Unable to add food details successfully. When I enter food details, such as the sample details as seen in the screenshot, and then click on save, a flash message is displayed saying, _You could not add a food item_ and entries are not added to food table. Kindly ensure to improve upon this aspect.

Kindly note, the **next code reviewer(s)** will be able to review other aspects of the requirement if you get this preliminary step working, and therefore there might be new issues highlighted in upcoming reviews.

**Error:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Team, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦   🚦  

## Good Points 👍  
- Descriptive Pull Request
- No linter errors


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 4 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/foods_controller.rb (hunk 1 above)
Comment (Gambit142) [reply app/controllers/foods_controller.rb:18]: Hi @rloterh, 

Thank you for the feedback

**Review by o-rashad — APPROVED** [review]
Asslam Alaikum Team @Gambit142 @MrBrN197 ,😊

Hi👋🏽, Your project is complete!  There is nothing else to say other than... it's time to merge it 😎 

# Congratulations! 🎉 

Happy coding!👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**